atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/DockerClientOrchestrationRule.java

```diff
@@ -0,0 +1,153 @@
+
+    private DockerComposeRule dockerComposeRule;
+    private DockerProxyRule dockerProxyRule;
+    private File configFile;
+
+    public DockerClientOrchestrationRule(TemporaryFolder temporaryFolder) {
+        this.temporaryFolder = temporaryFolder;
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        try {
+            configFile = temporaryFolder.newFile("atlasdb-ete.yml");
+            updateClientConfig(EMBEDDED_CONFIG);
+        } catch (IOException e) {
+            throw Throwables.propagate(e);
+        }
+
+        DockerMachine dockerMachine = createDockerMachine();
+        dockerComposeRule = DockerComposeRule.builder()
+                .machine(dockerMachine)
+                .file("docker-compose.timelock-migration.cassandra.yml")
+                .waitingForService("cassandra", HealthChecks.toHaveAllPortsOpen())
+                .saveLogsTo(LogDirectory.circleAwareLogDirectory(TimeLockMigrationEteTest.class.getSimpleName()))
+                .addClusterWait(new ClusterWait(ClusterHealthCheck.nativeHealthChecks(), WAIT_TIMEOUT))
+                .build();
+        dockerProxyRule = new DockerProxyRule(dockerComposeRule.projectName(), TimeLockMigrationEteTest.class);
+
+        dockerComposeRule.before();
+        dockerProxyRule.before();
+    }
+
+    @Override
+    protected void after() {
+        dockerProxyRule.after();
+        dockerComposeRule.after();
+    }
+
+    public void updateProcessLivenessScript() {
+        // Go-Java-Launcher's functionality for seeing if a process is running is not correct with respect to busybox.
+        // See also https://github.com/palantir/sls-packaging/issues/185
+        dockerExecOnClient("sed", "-i", "s/ps $PID > \\/dev\\/null;/kill -0 $PID/", "service/bin/init.sh");
+    }
+
+    public void updateClientConfig(File file) {
+        try {
+            FileUtils.writeStringToFile(configFile, FileUtils.readFileToString(file));
+        } catch (IOException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    public void restartAtlasClient() {
+        // Need nohup - otherwise our process is a child of our shell, and will be killed when we're done.
+        dockerExecOnClient("bash", "-c", "nohup service/bin/init.sh restart");
+    }
+
+    public String getClientLogs() {
+        return dockerExecOnClient("cat", "var/log/atlasdb-ete-startup.log");
+    }
+
+    private DockerMachine createDockerMachine() {
+        return DockerMachine.localMachine()
+                .withEnvironment(getEnvironment())
+                .build();
+    }
+
+    private Map<String, String> getEnvironment() {
+        return ImmutableMap.<String, String>builder()
+                .putAll(CassandraVersion.getEnvironment())
+                .put("CONFIG_FILE_MOUNTPOINT", temporaryFolder.getRoot().getAbsolutePath())
+                .build();
+    }
+
+    private String dockerExecOnClient(String... arguments) {
```

Comment:
Could this be shared with the e.g. nodetool repair and other cases where we use the ete container?